pypeit/core/procimg.py

```diff
@@ -672,9 +672,8 @@ def subtract_overscan(rawframe, datasec_img, oscansec_img, method='savgol', para
             osvar = np.pi/2*(np.sum(osvar)/osvar.size**2 if method.lower() == 'median' 
                              else np.sum(osvar, axis=compress_axis)/osvar.shape[compress_axis]**2)
         if method.lower() == 'polynomial':
-            # TODO: Use np.polynomial.polynomial.polyfit instead?
-            c = np.polyfit(np.arange(osfit.size), osfit, params[0])
-            ossub = np.polyval(c, np.arange(osfit.size))
+            poly = np.polynomial.Chebyshev.fit(np.arange(osfit.size), osfit, params[0])
+            ossub = poly(np.arange(osfit.size))
```

Comment:
I added the suggested code.  When I run this with `method=polynomial`, however, I am not seeing the `DeprecationWarning` (_i.e._, it is being filtered somewhere).  I couldn't find a warning filter within PypeIt, so it may be coming from one of the imported libraries.  In this case, do we change this to a `msgs.warn()`, or just let it be for now?  I've included the same notice in the `pypeitpar` listing, so it will show up in the docs.

Reply:
D'Oh!  Ah, well.  Those warnings would likely fly by unnoticed anyway.  I think having the notice in the docs is likely more useful anyway!